json-jackson-v2/src/main/java/tech/ydb/yoj/repository/db/json/JacksonJsonConverter.java

```diff
@@ -0,0 +1,98 @@
+package tech.ydb.yoj.repository.db.json;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.base.Suppliers;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import tech.ydb.yoj.repository.db.common.JsonConverter;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.function.Supplier;
+
+/**
+ * {@link JsonConverter YOJ JSON Converter} implementation using Jackson as the underlying JSON library.
+ * Use it to support JSON-valued fields ({@link tech.ydb.yoj.databind.schema.Column @Column(flatten=false)} composite
+ * objects and dynamic fields with type of interface/abstract class, e.g. {@link java.util.List}):
+ * <blockquote>
+ * <pre>
+ * CommonConverters.defineJsonConverter(new JacksonJsonConverter());
+ * </pre>
+ * </blockquote>
+ * <p>
+ * To customize the {@link ObjectMapper} being created, create a subclass of {@code JacksonJsonConverter} and override
+ * its {@link #createObjectMapper()} method.
+ */
+public class JacksonJsonConverter implements JsonConverter {
+    private final Supplier<ObjectMapper> mapper = Suppliers.memoize(this::createObjectMapper);
+
+    @Override
+    @SneakyThrows
```

Comment:
issue: @SneakyThrows is prohibited in proction-ready code. This would be nice to do with explicit exception catching

Reply:
Here the risk of `IOException`s escaping is little: the library uses `@SneakyThrows` already and is ready catch any exceptions from `JsonConverter`. Any thrown exceptions are then wrapped in `ConversionException`.

If (and when) we decide to change `JsonConverter`'s `throws` clauses, we can remove the `@SneakyThrows`.

Reply:
Maybe we can also start throwing unchecked `ConversionException` from here and annotate `JsonConverter` accordingly. I think there is some logic that prevents wrapping `ConversionException` inside `ConversionException` :-D

Reply:
Fixed